cmd/panfs-multipart.go

```diff
@@ -891,14 +992,41 @@ func (fs *PANFSObjects) AbortMultipartUpload(ctx context.Context, bucket, object
 		return toObjectErr(err, bucket, object)
 	}
 
+	fs.appendFileMapMu.Lock()
+	// Remove file in tmp folder
+	file := fs.appendFileMap[uploadID]
+	if file == nil {
+		file = fs.getPanFSMultipartAppendFile(bucketPath, object, uploadID)
+		fs.appendFileMap[uploadID] = file
+	}
+	fs.appendFileMapMu.Unlock()
+
+	file.flock.Lock()
```

Comment:
i would verify that uploadIDDir still exists after lock. Actually, I would even swap these two actions to something like (including my previous comment):

```
	fs.appendFileMapMu.Lock()
	// Remove file in tmp folder
	file := fs.appendFileMap[uploadID]
	if file == nil {
		file = fs.getPanFSMultipartAppendFile(bucketPath, object, uploadID)
		fs.appendFileMap[uploadID] = file
	}
	fs.appendFileMapMu.Unlock()

        err = file.flock.Lock()
        if err == nill {
            err = fsStatFile(ctx, pathJoin(uploadIDDir, fs.metaJSONFile))
        }

	if err != nil {
		if err == errFileNotFound || err == errFileAccessDenied || err == errCannotCreateFind {
			return InvalidUploadID{Bucket: bucket, Object: object, UploadID: uploadID}
		}
		return toObjectErr(err, bucket, object)
	}
```